src/main/java/collab/DataBase.java

```diff
@@ -2,84 +2,92 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
 import java.util.stream.Collectors;
-import collab.Employee;
 
 public class DataBase {
-  private ArrayList<Employee> employeeData = new ArrayList<Employee>();
+  HashSet<String> registerEmployeeNum = new HashSet<String>();
+  public ArrayList<Employee> employeeData = new ArrayList<Employee>();
   
   public ArrayList<Employee> getEmployeeAllData() {
     return employeeData;
   }
   
   public void add(Employee employee) {
-    employeeData.add(employee);        
+    if(registerEmployeeNum.contains(employee.getEmployeeNumber())) {
+      System.out.println(employee.getEmployeeNumber() + "Register Fail!! Employee Data Duplicate!!");
+      return;
+    }
+    registerEmployeeNum.add(employee.getEmployeeNumber());
+    employeeData.add(employee);
   }
-
+  
+  
   public Employee searchItem(String id) {
-    return searchItems("employeeNum", id).get(0);
+    List<Employee> foundItems = searchItems("employeeNum", id); 
+    if(foundItems.size()==0) return null;
```

Comment:
SonarCloud에 Code Smell 추천항목을 보니까 아래 처럼 변경하는 것을 추천하네요.

```suggestion
    if(foundItems.isEmpty()) return null;
```